### ISSUE TITLE
The Impact of the number of classes on detection performance.

### ISSUE THREAD

**Issue by anamozov** [AlexeyAB/darknet#3777]
I am working on license plate detection problem. My primary goal is detect a licence plate regardless of shape, color and etc.   When I divide license plates by shape or color to three classes, with the same training and testing data I achieve different mAP:
1. mAP with 1 class ~= 94%
2. mAP with 3 classes ~= 98%.
Is it normal? if Yes, what is the reason?

**Comment by LukeAI** [comment]
v. interesting - how big is your train/val dataset?

**Comment by LukeAI** [comment]
can you share the chart.png for the two runs?

**Comment by anamozov** [comment]
> v. interesting - how big is your train/val dataset?

Sorry for late answer. Number of images in train/val dataset: 1170/100.
Unfortunately, I did not save chart.png files. But both train and validation lines were nearly similar.

**Comment by LukeAI** [comment]
this is quite a small dataset, I suspect that the difference is just random fluctuation. I wanted to see the charts to see if there was overfitting or something like that.

**Comment by anamozov** [comment]
> I suspect that the difference is just random fluctuation

I think the same after testing both cases with new more complex data. Results were different, where one class network shows higher precision. 

> I wanted to see the charts to see if there was overfitting or something like that

I don't think that there was everfitting, because I chose the  .weights files which showed the best validation accuracy, while training error indicators remained almost unchanged at the lowest point, in both cases. 

And the reason why I opened this issue is to make sure if the number of classes really has an influence on the performance, because I have to decide what approach I will use before diving into annotating tens thousands of images.

**Comment by anamozov** [comment]
@AlexeyAB I am still wondering if the number of classes has an impact on mAP. Would be great if we have your opinion/comment on this question. Thank you in advance.
